src/axom/core/Array.hpp

```diff
@@ -757,12 +1203,41 @@ inline void Array< T >::dynamicRealloc( IndexType new_num_tuples )
     utilities::processAbort();
   }
 
-  m_data = axom::reallocate( m_data, new_capacity * m_num_components );
+  m_data = axom::reallocate< T >( m_data, new_capacity, m_allocator_id );
   m_capacity = new_capacity;
 
   assert( m_data != nullptr || m_capacity <= 0 );
 }
 
+//------------------------------------------------------------------------------
+/// Free functions implementing Array's operator(s)
+//------------------------------------------------------------------------------
+template< typename T >
+std::ostream& operator<<( std::ostream& os, const Array< T >& arr )
+{
+  arr.print(os);
+  return os;
+}
+
+template < typename T >
+bool operator==(const Array< T >& lhs, const Array< T >& rhs)
```

Comment:
How is this supposed to work on GPUs? More interestingly, what if `lhs` and `rhs` are in different memory spaces?

NOTE:
--------
1. For a given Array object, we can get the allocator ID and from that we could subsequently determine the execution space (we probably need some more traits mojo for that, but, it is feasible)
2. We could in theory bring move data around such that  both `lhs` and `rhs` are in the both memory space, but, I would strongly advise against that, just from a performance standpoint.

Perhaps, the thing to do is to throw an error if the two arrays are in a different memory space and just not allow for that operation? Thoughts?

Reply:
I think `operator==()` only makes sense if `lhs` and `rhs` are in the same memory space.  It makes sense to me to either return false or throw an error if they're in different memory spaces.